src/main/mainWindow.ts

```diff
@@ -419,18 +419,20 @@ function createMainWindow() {
                 backgroundMaterial: transparencyOption
             }),
         // Fix transparencyOption for custom discord titlebar
-        ...(customTitleBar &&
+        ...(titleBar === "custom" &&
             transparencyOption &&
             transparencyOption !== "none" && {
                 transparent: true
             }),
         ...(staticTitle && { title: "Vesktop" }),
-        ...(process.platform === "darwin" && getDarwinOptions()),
+        ...(process.platform === "darwin" && titleBar !== "shown" && getDarwinOptions()), // Show/Hide titlebar depending on settings on Mac
```

Comment:
getDarwinOptions() is responsible for more than just the titlebar so this will break various features